src/server/http.rs

```diff
@@ -3,19 +3,22 @@
 //! This contains some utilities to make using `reqwest` easier, including getting
 //! the correct TLS certificate store.
 
-use std::time::Duration;
-
 use crate::errors::Result;
 
-#[cfg(not(any(feature = "tls-native-roots", feature = "tls-webpki-roots")))]
+#[cfg(all(
+    not(target_arch = "wasm32"),
+    not(any(feature = "tls-native-roots", feature = "tls-webpki-roots"))
+))]
 compile_error!(
     "Either feature \"tls-native-roots\" or \"tls-webpki-roots\" must be enabled for HTTP client support."
 );
 
 static USER_AGENT: &str = concat!(env!("CARGO_PKG_NAME"), "/", env!("CARGO_PKG_VERSION"),);
 
 /// Create a new [`reqwest::Client`] with configuration appropriate to this library.
+#[cfg(not(target_arch = "wasm32"))]
 pub(super) fn client() -> Result<reqwest::Client> {
+    use std::time::Duration;
```

Comment:
I'm also curious if conditionally compiling `use` declarations actually results in smaller builds? Seems like the compiler would figure out that it isn't referenced.